share/normalize/links.py

```diff
@@ -368,7 +409,37 @@ def execute(self, obj):
 
 class TextLink(AbstractLink):
     def execute(self, obj):
-        return obj.text
+        if isinstance(obj, str):
+            return obj
+        elif isinstance(obj, dict):
+            if '#text' in obj:
+                return obj['#text']
+            raise Exception('#text is not in {}'.format(obj))
+        elif isinstance(obj, list):
+            for item in obj:
+                if isinstance(item, str):
+                    return item
```

Comment:
This will return the first string found in a list. Is that the expected functionality?